<launch>
    <!-- Turtlesim Node-->
    <node pkg="turtlesim" type="turtlesim_node" name="sim"/>
    <node pkg="turtlesim" type="turtle_teleop_key" name="teleop" output="log"/>

    <node pkg="rosservice" type="rosservice" name="create_turtle2" args='call --wait /spawn "{x: 2.0,y: 4.0, theta: 0.0, name: turtle2}" '/>
    <node pkg="rosservice" type="rosservice" name="create_turtle3" args='call --wait /spawn "{x: 4.0,y: 1.0, theta: 0.0, name: turtle3}" '/>

    <node name="turtle_controller1" pkg="srd_exercises" type="turtle_controller.py" respawn="false" output="screen" >
      <param name="turtlename" type="string" value="turtle1"/>
      <param name="sat_v" type="double" value="1.0"/>
      <param name="sat_w" type="double" value="1.0"/>
    </node>
    <node name="turtle_controller2" pkg="srd_exercises" type="turtle_controller.py" respawn="false" output="screen" >
      <param name="turtlename" type="string" value="turtle2"/>
      <param name="sat_v" type="double" value="1.0"/>
      <param name="sat_w" type="double" value="1.0"/>
    </node>
    <node name="turtle_controller3" pkg="srd_exercises" type="turtle_controller.py" respawn="false" output="screen" >
      <param name="turtlename" type="string" value="turtle3"/>
      <param name="sat_v" type="double" value="1.0"/>
      <param name="sat_w" type="double" value="1.0"/>
    </node>

    <node name="turtle_apf_cmd" pkg="srd_exercises" type="turtle_apf_cmd.py" respawn="false" output="screen" >
      <param name="turtle_num" type="int" value="3"/>
    </node>
    
</launch>
